source/jormungandr/jormungandr/scenarios/helper_classes/fallback_durations.py

```diff
@@ -100,6 +101,12 @@ def _get_duration(self, resp, place):
         }
         return map_response[resp.routing_status]
 
+    @new_relic.distributedEvent("routing_matrix", "street_network")
+    def _get_street_network_routing_matrix(self, origins, destinations):
+        return self._instance.get_street_network_routing_matrix(
```

Comment:
Can't you decorate instance method (here `get_street_network_routing_matrix`) directly?

Reply:
Nope, because it's a function that depends on the `_instance`. I don't want to decorate all the functions from every instances that implement `get_street_network_routing_matrix`. Even more, if I decorate every `get_street_network_routing_matrix` we will have to remember doing it the next instances that will be developed later... 

It's related to the open/close principle - https://en.wikipedia.org/wiki/Open%E2%80%93closed_principle